File: BackEnd/src/domain/dtos/auth/register-request.dto.ts
/**
 * @file register-request.dto.ts
 * @description DTO de solicitud de registro de usuario con validaciones class-validator.
 * @module Auth
 * @layer Domain
 */

import { IsEmail, IsString, MinLength, MaxLength } from 'class-validator';

/**
 * DTO de registro. Validado en el controlador antes de llegar al use case.
 *
 * [⚠ TFG]: Los decoradores class-validator en DTOs de dominio son una
 * concesión de pragmatismo: en Clean Architecture pura los DTOs de dominio
 * no deberían depender de un framework de validación, pero se acepta el
 * acoplamiento por simplicidad para el alcance del proyecto.
 */
export class RegisterRequestDTO {
  /** Nombre completo del usuario */
  @IsString()
  @MinLength(2)
  @MaxLength(100)
  name!: string;

  /** Email único del usuario */
  @IsEmail()
  email!: string;

  /** Contraseña en texto plano (se hashea antes de persistir) */
  @IsString()
  @MinLength(8)
  @MaxLength(128)
  password!: string;
}

File: FrontEnd/lib/domain/dtos/auth/register_request_dto.dart
/// @file register_request_dto.dart
/// @description DTO de entrada para el caso de uso de registro de usuario.
/// Encapsula y valida los datos del formulario antes de pasarlos al use case.
/// @module Core
/// @layer Domain
library;

// ═══════════════════════════════════════════════════════════════════════════════
// REGISTER REQUEST DTO
// ═══════════════════════════════════════════════════════════════════════════════

/// DTO de registro de usuario. Usado por el ViewModel y el use case de registro.
class RegisterRequestDto {
  const RegisterRequestDto({
    required this.name,
    required this.email,
    required this.password,
  });

  /// Nombre visible del nuevo usuario.
  final String name;

  /// Email del nuevo usuario (debe ser único en el sistema).
  final String email;

  /// Contraseña en texto plano (el backend la hashea con bcrypt).
  final String password;
}
